sdk/python/kfp/_local_client.py

```diff
@@ -168,13 +169,14 @@ def images_to_exclude(self):
         def ops_to_exclude(self):
             return self._ops_to_exclude
 
-    def __init__(self, pipeline_root: str = "/tmp") -> None:
+    def __init__(self, pipeline_root: str = None) -> None:
         """Construct the instance of LocalClient
 
         Args：
             pipeline_root: The root directory where the output artifact of component
               will be savad.
         """
+        pipeline_root = pipeline_root or tempfile.tempdir
```

Comment:
Shall we consider warning users when this is default to None? Because one of the reason we have pipeline root is to ensure the artifacts generated during the pipeline execution get persistent somewhere, and using `tempfile` is 'against' this purpose

Reply:
Interesting idea. I can try adding a warning.
Although in this PR I'm not changing the behavior - just changing the default from `/tmp` to the portable `tempdir`.